sdk/src/test/java/io/customer/sdk/CustomerIOTest.kt

```diff
@@ -156,6 +158,55 @@ internal class CustomerIOTest : BaseTest() {
         verifySuccess(response, Unit)
     }
 
+    @Test
+    fun `verify SDK both default and custom attributes are added`() {
+
+        `when`(
+            mockCustomerIO.api.registerDeviceToken(
+                any(),
+                any(),
+            )
+        ).thenReturn(getEmptyAction())
+
+        customerIO.deviceAttributes.putAll(mapOf("test" to "value"))
+
+        val expectedToken = "token"
+
+        val expectedAttributes = customerIO.deviceAttributes + deviceStore.buildDeviceAttributes()
```

Comment:
I have an idea to improve the quality of this test. 

For the expected value, have as little logic involved in generating the expected value is a good idea. What I mean by logic is:
```kt
// logic is involved to create the expected value
val expected = 2 + 2 // <-- if this had a bug in it, the test could give a false positive

// compared to no logic
val expected = 4 // <-- no logic == less chance of encountering a false positive. 
```

In my past experience, I have encountered times where my test functions gave false positives because what I truly expected to be the expected value was incorrect. That was because it had too much logic involved where the logic actually had a bug in it. 

So, I would suggest that this line of code instead says:
```kt
// instead of having logic generate the expected value, hard-code the expected value in. 
val expectedAttributes = mapOf("test" to "value, "push_enabled" to "true)
```

yes, this would require modifying the returned value from `deviceStore.buildDeviceAttriutes()` to something hard-coded/predictable but stubs should give the ability to modify returned values so it shouldn't be a problem.